### PR TITLE
Add feature to allow user to choose number of comments to display

### PR DESCRIPTION
**Pull request by haoyxc**
Add a select menu to index that lets user decide how many comments to see and displays at most the number. 

Added a query parameter when making a request to the server so that the server only queries up to that many. Removed the work of slicing the queried results from the client.

image
image

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -17,42 +17,34 @@
...
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    List<Comment> data = new ArrayList<>(); 
-    
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {  
+    String numParam = request.getParameter("num");
```
Comment (fenghaolw) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:38]: Again, what if "num" is not presented :)

Comment (haoyxc) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:38]: would it be appropriate to do something similar (and just return a 400?)

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -55,16 +55,21 @@ function loadSkills () {
  * Makes a request to /data and gets the response
  */
 async function fetchFromData() {
-  const response = await fetch('/data');
+  const selector = $("#num-comments")[0]; //get the select element
+  const selectedNum = selector.options[selector.selectedIndex].value; //number of comments user wants
```
Comment (EatingW) [inline portfolio/src/main/webapp/script.js]: I think you can just do $("#num-comments").val() to get the value of the selected element.
https://api.jquery.com/val/

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -55,16 +55,21 @@ function loadSkills () {
...
+
+  //make request with the number specified
+  const response = await fetch(`/data?num=${selectedNum}`);
   
-  // This gives a list of messages
+  // This gives a list of comments
   const comments = await response.json();
+  const numComments = comments.length;
```
Comment (EatingW) [inline portfolio/src/main/webapp/script.js]: Looks like you never used numComments anywhere?

**Review by EatingW — CHANGES_REQUESTED** [review]

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (haoyxc) [reply portfolio/src/main/webapp/script.js]: oh my gosh I feel so dumb thank you!

**Review by EatingW — APPROVED** [review]
LGTM!

**Review by fenghaolw — APPROVED** [review]

**Review by fenghaolw — APPROVED** [review]

**Pull request merged**